addons/auth_signup/controllers/main.py

```diff
@@ -115,10 +115,13 @@ def get_auth_signup_qcontext(self):
                 qcontext['error'] = _("Invalid signup token")
                 qcontext['invalid_token'] = True
         return qcontext
+    
+    def _allowed_field_signup(self):
+        return ('login', 'name', 'password')
 
     def do_signup(self, qcontext):
         """ Shared helper that creates a res.partner out of a token """
-        values = { key: qcontext.get(key) for key in ('login', 'name', 'password') }
+        values = { key: qcontext.get(key) for key in self._allowed_field_signup() }
```

Comment:
Please retarget to master as you suggested ;)

Also it seems the refactoring would be more flexible if you moved the whole logic of preparing `values` into a `_prepare_signup_values()`, giving extensions more freedom on what they wish to do, rather than a method only returning a flat list of keys.

```suggestion
        values = self._prepare_signup_values(qcontext)
```

i.e basically:

```py
    def _prepare_signup_values(self, qcontext):
        values = { key: qcontext.get(key) for key in ('login', 'name', 'password') }
        if not values:
            raise UserError(_("The form was not properly filled in."))
        if values.get('password') != qcontext.get('confirm_password'):
            raise UserError(_("Passwords do not match; please retype them."))
        supported_lang_codes = [code for code, _ in request.env['res.lang'].get_installed()]
        lang = request.context.get('lang', '')
        if lang in supported_lang_codes:
            values['lang'] = lang
        return values

    def do_signup(self, qcontext):
        values = self._prepare_signup_values(qcontext)
        self._signup_with_values(qcontext.get('token'), values)
        equest.env.cr.commit()
```

This way your extension only needs to do this:

```py
    def _prepare_signup_values(self, qcontext):
        values = super()._prepare_signup_values(qcontext)
        values['foo'] = qcontext.get('foo')
        return values
```